### PR TITLE
Add endpoint to get directory uuid from directory path.

### PR DESCRIPTION
**Pull request by FranckLecuyer**

### CODE REVIEW

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java [hunk 1]
```diff
@@ -263,4 +263,13 @@ public ResponseEntity<Page<DirectoryElementInfos>> searchElements(
         return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON)
                 .body(service.searchElements(userInput, directoryUuid));
     }
+
+    @GetMapping(value = "/directories/uuid", consumes = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(summary = "Get uuid from directory described by the path")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "The directory uuid"),
+    })
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/directory/server/DirectoryController.java:271]: Add @ApiResponse 404 not found here

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java [hunk 2]
```diff
@@ -263,4 +263,13 @@ public ResponseEntity<Page<DirectoryElementInfos>> searchElements(
         return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON)
                 .body(service.searchElements(userInput, directoryUuid));
     }
+
+    @GetMapping(value = "/directories/uuid", consumes = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(summary = "Get uuid from directory described by the path")
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/directory/server/DirectoryController.java:268]:
```suggestion
    @Operation(summary = "Get directory uuid from given path")
```

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java [hunk 3]
```diff
@@ -263,4 +263,13 @@ public ResponseEntity<Page<DirectoryElementInfos>> searchElements(
         return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON)
                 .body(service.searchElements(userInput, directoryUuid));
     }
+
+    @GetMapping(value = "/directories/uuid", consumes = MediaType.APPLICATION_JSON_VALUE)
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/directory/server/DirectoryController.java:267]: this endpoint doesn't consume JSON_VALUE

File: src/test/java/org/gridsuite/directory/server/DirectoryTest.java [hunk 4]
```diff
@@ -1800,6 +1800,59 @@ public void testCountUserCases() throws Exception {
...
+        UUID resultUuid = objectMapper.readValue(mvcResult.getResponse().getContentAsString(), UUID.class);
+        assertEquals(resultUuid, subDirAttributes4.getElementUuid());
+        output.clear();
+
+        // unexisting directory
+        mockMvc
+            .perform(get("/v1/directories/uuid?directoryPath=" + "root/dir1/dir5")
+                .contentType(MediaType.APPLICATION_JSON))
+            .andExpectAll(status().isNotFound()).andReturn();
+        output.clear();
```
Comment (sBouzols) [inline src/test/java/org/gridsuite/directory/server/DirectoryTest.java:1853]: Could we add a unit test here for the case we give a path to an element not a directory ?
should return isNotFound() as well

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java [hunk 5]
```diff
@@ -263,4 +263,13 @@ public ResponseEntity<Page<DirectoryElementInfos>> searchElements(
...
+    @Operation(summary = "Get uuid from directory described by the path")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "The directory uuid"),
+    })
+    public ResponseEntity<UUID> getDirectoryUuidFromPath(@RequestParam("directoryPath") String directoryPath) {
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/directory/server/DirectoryController.java:272]: We should decode path here, same model as for this commit :
https://github.com/gridsuite/directory-server/commit/4d88e0df0ae99939ebf23a1645ed98bafc7f51b9

File: src/test/java/org/gridsuite/directory/server/DirectoryTest.java [hunk 6]
```diff
@@ -1800,6 +1800,59 @@ public void testCountUserCases() throws Exception {
...
+    }
```
Comment (sBouzols) [inline src/test/java/org/gridsuite/directory/server/DirectoryTest.java:1854]: Add a test with an encoded path please with %... see https://github.com/gridsuite/directory-server/commit/4d88e0df0ae99939ebf23a1645ed98bafc7f51b9

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java (hunk 1 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/directory/server/DirectoryController.java]: :+1:

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java (hunk 2 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/directory/server/DirectoryController.java]: :+1:

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java (hunk 3 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/directory/server/DirectoryController.java]: :+1:

File: src/test/java/org/gridsuite/directory/server/DirectoryTest.java (hunk 4 above)
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/directory/server/DirectoryTest.java:1860]: Done

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java (hunk 5 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/directory/server/DirectoryController.java:271]: :+1:  Done

File: src/test/java/org/gridsuite/directory/server/DirectoryTest.java (hunk 6 above)
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/directory/server/DirectoryTest.java:1877]: :+1:  Done

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java (hunk 1 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/directory/server/DirectoryController.java]: Done

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java (hunk 2 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/directory/server/DirectoryController.java]: Done

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java (hunk 3 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/directory/server/DirectoryController.java]: Done

**Review by sBouzols — APPROVED** [review]
Code review OK
Tests OK partially, will fix "/" case in another PR

* test folder name with withspace = GET uuid OK :heavy_check_mark: 
* test folder name with "/" = GET uuid KO :warning: 
* test folder name with "\" -> we can't create this folder in GridExplore
* test folder name with "éùàèöïâ" = GET uuid OK :heavy_check_mark: 
* test root folder name with withspace = GET uuid OK :heavy_check_mark: 
* test root folder name with "/" = GET uuid KO :warning: 
* test root folder name with "\" -> ça on peut apparemment = GET uuid OK :heavy_check_mark: 
* test root folder name with "éùàèöïâ" = GET uuid OK :heavy_check_mark:

**Review by dbraquart — APPROVED** [review]
code: ok
tests: ok

**Review by ayolab — APPROVED** [review]

**Pull request merged**